src/BloomBrowserUI/bookEdit/OverflowChecker/OverflowChecker.ts

```diff
@@ -88,28 +93,37 @@ class OverflowChecker {
 
     // Actual testable determination of Type II overflow or not
     // 'public' for testing (2 types of overflow are defined in MarkOverflowInternal below)
-    public static IsOverflowingMargins(element: HTMLElement) : boolean {
+    // returns nearest ancestor that this element overflows
+    public static overflowingAncestor(element: HTMLElement) : HTMLElement {
         // Ignore Topic divs as they are chosen from a list
         if (element.hasAttribute('data-book') && element.getAttribute('data-book') == "topic") {
-            return false;
+            return null;
         }
-        // We want to prevent an inner div from expanding past the borders set by any containing marginBox class.
-        var marginBoxParent = $(element).parents('.marginBox');
-        var parentBottom;
-        if(marginBoxParent && marginBoxParent.length > 0)
-            parentBottom = $(marginBoxParent[0]).offset().top + $(marginBoxParent[0]).outerHeight(true);
-        else
-            parentBottom = 999999;
-        var elemTop = $(element).offset().top;
-        var elemBottom = elemTop + $(element).outerHeight(false);
-        // console.log("Offset top: " + elemTop + " Outer Height: " + $(element).outerHeight(false));
-        // If css has "overflow: visible;", scrollHeight is always 2 greater than clientHeight.
-        // This is because of the thin grey border on a focused input box.
-        // In fact, the focused grey border causes the same problem in detecting the bottom of a marginBox
-        // so we'll apply the same 'fudge' factor to both comparisons.
-        var focusedBorderFudgeFactor = 2;
-
-        return  elemBottom > parentBottom + focusedBorderFudgeFactor;
+        // We want to prevent an inner div from expanding past the borders set by any fixed containing element.
+        var parents = $(element).parents();
+        if (!parents || parents.length == 0) {
+            return null;
+        }
+        for(var i=0; i < parents.length; i++) { // search ancestors starting with nearest
+            var $this = $(parents[i]);
```

Comment:
Why do you want to use a name already as overworked as 'this' for the current parent?